packages/components/src/color-palette/index.native.js

```diff
@@ -0,0 +1,190 @@
+
+	useEffect( () => {
+		scrollViewRef.current.scrollTo( { x: 0, y: 0 } );
+	}, [ currentSegment ] );
+
+	function isSelectedCustom() {
+		const colors = isGradientSegment
+			? defaultGradientColors
+			: defaultColors;
+		return ! isGradient && activeColor && ! colors.includes( activeColor );
+	}
+
+	function isSelected( color ) {
+		return ! isSelectedCustom() && activeColor === color;
+	}
+
+	function timingAnimation( property, toValue ) {
+		return Animated.timing( property, {
+			toValue,
+			duration: ANIMATION_DURATION,
+			easing: Easing.ease,
+		} );
+	}
+
+	function performAnimation( color ) {
+		opacity.setValue( isSelected( color ) ? 1 : 0 );
+		scale.setValue( 1 );
+
+		Animated.parallel( [
+			timingAnimation( scale, 2 ),
+			timingAnimation( opacity, 1 ),
+		] ).start();
+	}
+
+	const scaleInterpolation = scale.interpolate( {
+		inputRange: [ 1, 1.5, 2 ],
+		outputRange: [ 1, 0.7, 1 ],
+	} );
+
+	function onColorPress( color ) {
+		performAnimation( color );
+		setColor( color );
+	}
+
+	function Palette( { isGradientPalette, withCustomPicker } ) {
+		const palette = isGradientPalette
+			? defaultGradientColors
+			: defaultColors;
+		const verticalSeparatorStyle = getStylesFromColorScheme(
+			styles.verticalSeparator,
+			styles.verticalSeparatorDark
+		);
+
+		return (
+			<>
+				{ palette.map( ( color ) => {
+					const scaleValue = isSelected( color )
+						? scaleInterpolation
+						: 1;
+					return (
+						<TouchableWithoutFeedback
+							onPress={ () => onColorPress( color ) }
+							key={ color }
+						>
+							<Animated.View
+								style={ {
+									transform: [
+										{
+											scale: scaleValue,
+										},
+									],
+								} }
+							>
+								<ColorIndicator
+									color={ color }
+									isSelected={ isSelected( color ) }
+									opacity={ opacity }
+									style={ styles.colorIndicator }
+								/>
+							</Animated.View>
+						</TouchableWithoutFeedback>
+					);
+				} ) }
+				{ withCustomPicker && (
+					<>
+						<View style={ verticalSeparatorStyle } />
+						<TouchableWithoutFeedback onPress={ onCustomPress }>
+							<View>
+								<ColorIndicator
+									withCustomPicker
+									color={ customSwatchGradients }
+									isSelected={ isSelectedCustom() }
+									style={ styles.colorIndicator }
+								/>
+							</View>
+						</TouchableWithoutFeedback>
+					</>
+				) }
+			</>
+		);
+	}
+
+	return (
+		<ScrollView
+			contentContainerStyle={ styles.contentContainer }
+			style={ styles.container }
+			horizontal
+			showsHorizontalScrollIndicator={ false }
+			keyboardShouldPersistTaps="always"
+			disableScrollViewPanResponder
+			onScrollBeginDrag={ () => shouldEnableBottomSheetScroll( false ) }
+			onScrollEndDrag={ () => shouldEnableBottomSheetScroll( true ) }
+			ref={ scrollViewRef }
+		>
+			<TouchableWithoutFeedback>
+				<Palette
+					withCustomPicker={ ! isGradientSegment }
```

Comment:
I think it would be enough to pass `isGradientSegnent`but I'm not sure if it needs to be passed at all. You have access to all these values inside the Pallete w/o needing to pass them as props. The `onCustomPress` is not even used :)